src/sql/base/browser/ui/table/plugins/cellSelectionModel.plugin.ts

```diff
@@ -118,6 +126,114 @@ export class CellSelectionModel<T> implements Slick.SelectionModel<T, Array<Slic
 		}
 	}
 
+
+	/**
+	 * DO NOT CALL THIS DIRECTLY - GO THROUGH INSERT INTO SELECTIONS
+	 *
+	 */
+	private mergeSelections(ranges: Array<Slick.Range>, range: Slick.Range) {
```

Comment:
Whats the perf of this like? does it have nay issues as  you have a lot of disjointed ranges?

Reply:
I haven't had performance issues personally. Where this could get problematic is if we have a ton of disjointed ranges. But since we're merging the disjointed ranges together, we end up with < 10 for most selections.

You'd have to hit over 100 disjointed ranges to be able to notice the additional time taken with this I believe. Although quick tests from my side shows its not really having an impact in these scenarios either.